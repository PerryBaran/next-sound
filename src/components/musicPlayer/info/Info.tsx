"use client"

import style from "./info.module.css"
import Image from "next/image"
import { music } from "@/media/icons"

interface Props {
  albumArt: string | undefined
  artistName: string | undefined
  songName: string | undefined
  albumName: string | undefined
}

export default function Info(props: Props) {
  const { albumArt, artistName, albumName, songName } = props

  return (
    <div className={style.container}>
      {albumArt ? (
        <Image
          src={albumArt}
          alt={`${albumName} cover art`}
          height={75}
          width={75}
        />
      ) : (
        <Image
          src={music}
          alt="default cover art"
          height={75}
          width={75}
          style={{ filter: "invert()" }}
        />
      )}

      <div>
        <h3>{artistName}</h3>
        <p>{songName}</p>
      </div>
    </div>
  )
}
