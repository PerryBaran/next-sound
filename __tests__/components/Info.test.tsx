import { render, screen } from "@testing-library/react"
import Info from "../../src/components/musicPlayer/info/Info"

describe("Info", () => {
  const props = {
    albumArt: "/art",
    artistName: "artist-name",
    songName: "song-name",
    albumName: "album-name"
  }

  test("renders correctly with defined props", () => {
    render(<Info {...props}/>)

    expect(screen.getByAltText(`${props.albumName} cover art`)).toHaveAttribute("src", `/_next/image?url=%2F${props.albumArt.substring(1, props.albumArt.length)}&w=256&q=75`)
    expect(screen.getByText(props.artistName)).toBeInstanceOf(HTMLHeadingElement)
    expect(screen.getByText(props.songName)).toBeInstanceOf(HTMLParagraphElement)
  })

  test("renders correctly with undefined props", () => {
    render(<Info />)

    expect(screen.getByAltText(/default cover art/i)).toBeInTheDocument()
  })

  describe("snapshots", () => {
    test("defined props", () => {
      const { asFragment } = render(<Info {...props}/>)

      expect(asFragment()).toMatchSnapshot()
    })

    test("undefined props", () => {
      const { asFragment } = render(<Info />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})