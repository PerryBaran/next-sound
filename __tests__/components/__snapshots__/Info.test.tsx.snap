// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info snapshots defined props 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <img
      alt="album-name cover art"
      data-nimg="1"
      decoding="async"
      height="75"
      loading="lazy"
      src="/_next/image?url=%2Fart&w=256&q=75"
      srcset="/_next/image?url=%2Fart&w=96&q=75 1x, /_next/image?url=%2Fart&w=256&q=75 2x"
      style="color: transparent;"
      width="75"
    />
    <div
      class="info"
    >
      <h3>
        artist-name
      </h3>
      <p>
        song-name
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Info snapshots undefined props 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <img
      alt="default cover art"
      data-nimg="1"
      decoding="async"
      height="75"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
      style="color: transparent; filter: invert();"
      width="75"
    />
    <div
      class="info"
    >
      <h3 />
      <p />
    </div>
  </div>
</DocumentFragment>
`;
